app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java

```diff
@@ -149,17 +144,20 @@ public String getAppLanguage() {
     public void onFragmentFirstTimeVisible() {
     }
 
-    @Override
-    public void setUserVisibleHint(boolean isVisibleToUser) {
-        super.setUserVisibleHint(isVisibleToUser);
-        synchronized (_fragmentFirstTimeVisibleSync) {
-            if (isVisibleToUser && _fragmentFirstTimeVisible) {
-                _fragmentFirstTimeVisible = false;
-                onFragmentFirstTimeVisible();
-            }
+    private synchronized void checkRunFirstTimeVisible() {
+        if (_fragmentFirstTimeVisible && isVisible() && isResumed()) {
+            _fragmentFirstTimeVisible = false;
+            onFragmentFirstTimeVisible();
         }
+    }
 
-        attachToolbarClickListenersToFragment();
+    @Override
+    public void onResume() {
+        super.onResume();
+        final View view = getView();
+        if (view != null) {
+            view.postDelayed(this::checkRunFirstTimeVisible, 1);
```

Comment:
May be good to bump up the delay to something higher. Something like ~ 200ms will be barely perceptible to user, but everything will have had time to draw and layout.

Reply:
stability over speed nitpickers :-)

Reply:
It's still blazing fast. I have not found an editor which opens and highlights a large file as fast as markor :)

Reply:
Can you remotly reproduce the permoance issue mentioned in #1833 with ~5 seconds? I do not.

Especially with the permission request change it got much quicker on first start and further starts as far I can tell. (unrelated to edior)

Reply:
> Can you remotly reproduce the permoance issue mentioned in https://github.com/gsantner/markor/issues/1833 with ~5 seconds? I do not.

My reading was that this is happening with the debug build... Which I expect to be much slower/